cpp/src/arrow/compute/kernels/vector_replace_test.cc

```diff
@@ -25,6 +25,8 @@
 #include "arrow/util/checked_cast.h"
 #include "arrow/util/key_value_metadata.h"
 #include "arrow/util/make_unique.h"
+#include <arrow/array/concatenate.h>
+#include <arrow/testing/generator.h>
```

Comment:
nit: use `#include ""` for consistency/use `""` when including Arrow headers inside the codebase